shell/platform/android/io/flutter/embedding/android/FlutterActivity.java

```diff
@@ -521,6 +618,27 @@ public FlutterShellArgs getFlutterShellArgs() {
     return FlutterShellArgs.fromIntent(getIntent());
   }
 
+  /**
+   * Returns the ID of a statically cached {@link FlutterEngine} to use within this
+   * {@code FlutterActivity}, or {@code null} if this {@code FlutterActivity} does not want to
+   * use a cached {@link FlutterEngine}.
+   */
+  @Override
+  @Nullable
+  public String getCachedEngineId() {
+    return getIntent().getStringExtra(EXTRA_CACHED_ENGINE_ID);
+  }
+
+  /**
+   * Returns false if the {@link FlutterEngine} backing this {@code FlutterActivity} should
+   * outlive this {@code FlutterActivity}, or true to be destroyed when the {@code FlutterActivity}
+   * is destroyed.
+   */
+  @Override
+  public boolean shouldDestroyEngineWithHost() {
+    return getIntent().getBooleanExtra(EXTRA_DESTROY_ENGINE_WITH_ACTIVITY, false);
```

Comment:
Describe the default either in this doc or in the builder class or the builder factory.

Reply:
Done. I also updated FlutterActivity and FlutterFragment to set this value to true whenever the FlutterActivity/Fragment create its own FlutterEngine, or one is explicitly provided by subclassing. Do you think that's OK?